### PR TITLE
Bug fixes

### LINKED ISSUE #205: Markdown image render fail when url contain param
Some image fail, it seem to happen when image url contain `?param=` See:
http://localhost:3000/food/@lily-da-vine/adventures-of-an-american-chef-on-the-run-fair-style-funnel-cakes
http://localhost:3000/censorship/@mynameisbrian/storing-images-on-the-steem-blockchain-censorship-resistant-images

### LINKED ISSUE #206: Delete category icon is not working
The button delete category isn't working
image

### PR DESCRIPTION
**Pull request by nil1511**
Fixed #205 
Fixed #206 

Fixed issue with lodash has functions not working properly. In future Check the feature set https://github.com/lodash/lodash-webpack-plugin#feature-sets and enabled accordingly if it does not work in your case.

Markdown fixes tested with
http://localhost:3000/steemit/@trevor.george/thinking-of-steemit-and-content-that-could-bring-in-a-new-wave-of-members
http://localhost:3000/spanish/@danilamarilu/veo-veo-que-ves-original-art
http://localhost:3000/steemitphotochallenge/@mynameisbrian/steemit-photo-challenge-entry-3-23-nostaligia-ned-head
http://localhost:3000/food/@lily-da-vine/adventures-of-an-american-chef-on-the-run-fair-style-funnel-cakes
http://localhost:3000/censorship/@mynameisbrian/storing-images-on-the-steem-blockchain-censorship-resistant-images

### CODE REVIEW

File: src/post/Body.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import React from 'react';
-import { has } from 'lodash';
+import _ from 'lodash';
 import steemembed from 'steemembed';
 import sanitizeHtml from 'sanitize-html';
 import Remarkable from 'remarkable';
 
 import sanitizeConfig from './../helpers/SanitizeConfig';
+import { replaceAll, escapeRegExp, imageRegex } from './../helpers/regexHelpers';
```
Comment (adcpm) [inline src/post/Body.js]: We should use ../ instead of ./../

File: src/post/Body.js [hunk 2]
```diff
@@ -23,42 +24,46 @@ export default (props) => {
...
 
-      const newUrl = `https://img1.steemit.com/0x0/${image}`;
-      body = body.replace(new RegExp(image, 'g'), newUrl);
-      body = body.replace(new RegExp(`<a href="${newUrl}">${newUrl}</a>`, 'g'), `<img src="${newUrl}">`);
+      body = replaceAll(body, `<a href="${image}">${image}</a>`, `<img src="${newUrl}">`);
       // not in img tag
-      if (body.search(`<img.+?src=["']${newUrl}["'].*?>`) === -1) {
-        body = body.replace(new RegExp(newUrl, 'g'), `<img src="${newUrl}">`);
+      if (body.search(`<img[^>]+src="${escapeRegExp(image)}"`) === -1) {
+        body = replaceAll(body, image, `<img src="${newUrl}">`);
```
Comment (adcpm) [inline src/post/Body.js]: The img tag should be closed ``

File: src/post/Write/Write.js [hunk 3]
```diff
@@ -105,7 +105,7 @@ export class RawNewPost extends Component {
       <span key={key} {...other}>
         {getTagDisplayValue(tag)}
         {!disabled &&
-          <Icon className={classNameRemove} onClick={() => onRemove(key)} name="close" />
+          <button onClick={() => onRemove(key)}><Icon className={classNameRemove} name="close" /></button>
```
Comment (adcpm) [inline src/post/Write/Write.js]: Can you use <a> link instead of button?

File: src/post/Body.js (hunk 2 above)
Comment (nil1511) [inline src/post/Body.js]: replace is happening on data processed by remarkable so tags will be closed. any matching for end bracket was making regex more complicated.

File: src/post/Write/Write.js (hunk 3 above)
Comment (nil1511) [inline src/post/Write/Write.js]: you can but schematically it will not be correct.

Comment (adcpm) [inline src/post/Write/Write.js]: I mean <a link to not having the fefault vutton styling

Comment (nil1511) [inline src/post/Write/Write.js]: yes. schematically a tag should not have onClick as it has href.

File: src/post/Body.js (hunk 2 above)
Comment (nil1511) [inline src/post/Body.js]: it should be as `` not as `` img tag does not need closing.

**Pull request merged**